test/test_xacro.py

```diff
@@ -1136,6 +1136,38 @@ def test_yaml_support(self):
             self.assert_matches(self.quick_xacro(src, cli=['type:=%s' % i]),
                                 res.format(tag=i))
 
+    def test_yaml_support_dotted(self):
+        src = '''
+<a xmlns:xacro="http://www.ros.org/wiki/xacro">
+  <xacro:property name="settings" value="${load_yaml('settings.yaml')}"/>
+  <xacro:property name="type" value="$(arg type)"/>
+  <xacro:include filename="${settings.arms[type].file}"/>
+  <xacro:call macro="${settings.arms[type].macro}"/>
+</a>'''
+        res = '''<a><{tag}/></a>'''
+        for i in ['inc1', 'inc2']:
+            self.assert_matches(self.quick_xacro(src, cli=['type:=%s' % i]),
+                                res.format(tag=i))
+
+    def test_yaml_support_dotted_key_error(self):
+        src = '''
+<a xmlns:xacro="http://www.ros.org/wiki/xacro">
+  <xacro:property name="settings" value="${load_yaml('settings.yaml')}"/>
+  <xacro:property name="bar" value="${settings.baz}"/>
+  ${bar}
+</a>'''
+        self.assertRaises(xacro.XacroException, self.quick_xacro, src)
+
+    def test_yaml_support_dotted_arith(self):
+        src = '''
+<a xmlns:xacro="http://www.ros.org/wiki/xacro">
+  <xacro:property name="settings" value="${load_yaml('settings.yaml')}"/>
+  <xacro:property name="bar" value="${settings.arms.inc2.props.port + 1}"/>
+  ${bar}
+</a>'''
+        res = '''<a>4243</a>'''
+        self.assert_matches(self.quick_xacro(src), res)
```

Comment:
Not too sure about this test, but I wanted to test at last *something* which uses the results of the `YamlDictWrapper` in an expression.